components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/engine/runtime/xpath/JaxpFunctionResolver.java

```diff
@@ -194,18 +245,24 @@ public Object evaluate(List args) throws XPathFunctionException {
     /**
      * Returns the potential owners of the task. It MUST evaluate to an empty htt:organizationalEntity in case of an
      * error. If the task name is not present the current task MUST be considered.
+     * In : task name (optional)
+     * Out : potential owners (htt:organizationalEntity)
      */
     public class GetPotentialOwners implements XPathFunction {
+
         public Object evaluate(List args) throws XPathFunctionException {
             GenericHumanRoleDAO potentialOwners = null;
+            TaskDAO taskDAO = evalCtx.getTask();
             if (args.size() == 0) {
                 // Case 1: consider current Task.
-                potentialOwners = evalCtx.getGenericHumanRole(
-                        GenericHumanRoleDAO.GenericHumanRoleType.POTENTIAL_OWNERS);
+                potentialOwners = getPotentialOwnersFromCtx();
             } else if ((args.size() == 1) && (args.get(0) instanceof String)) {
                 String taskName = (String) args.get(0);
-                if (!StringUtils.isNullOrEmpty(taskName)) {
-                    // Case 2: Getting specific task. Unsupported.
+                if (taskName.equals(taskDAO.getName())) {
+                    //Case 2: TaskName equals to current task, consider current task
+                    potentialOwners = getPotentialOwnersFromCtx();
+                } else if (!StringUtils.isNullOrEmpty(taskName)) {
```

Comment:
In the line 259, I check the args size before this. So that taskName cannot be null. Thank you.